plugins/participants.js

```diff
@@ -0,0 +1,167 @@
+module.exports = (controller) => {
+    const renderParticipantsBlocks = async (participants) => {
+        return Promise.all(
+            participants.map((participant) =>
+                renderParticipantBlocks(participant)
+            )
+        ).then((participants) => participants.flat());
+    };
+
+    const renderParticipantBlocks = async (participant) => {
+        const links = participant.links.map((link) => {
+            return {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: `:link: <${link.url}|${link.name}>`,
+                },
+            };
+        });
+        const gauges = participant.gauges.map((gauge) => {
+            return {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: participants.getGaugeString(gauge),
+                },
+            };
+        });
+        const details = participant.details.map((detail) => {
+            return {
+                type: "mrkdwn",
+                text: `*${detail.name}*: ${detail.value}`,
+            };
+        });
+        return [
+            {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: `:bust_in_silhouette: *${participant.info.real_name}*`,
+                },
+                accessory: {
+                    type: "image",
+                    image_url: participant.info.profile.image_72,
+                    alt_text: participant.info.real_name,
+                },
+                fields: details.length ? details : undefined,
+            },
+            ...gauges,
+            ...links,
+        ];
+    };
+
+    const renderNoContentBlocks = () => {
+        return [
+            {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text:
+                        "Nothing to display yet. Try adding some participant info.",
+                },
+            },
+        ];
+    };
+
+    const participants = {
+        name: "Participants",
+        init: () => {
+            controller.addPluginExtension("participants", participants);
+        },
+        getParticipant: async (userId) => {
+            const userInfo = controller.plugins.slack.getUserInfo(userId);
+            const details = controller.plugins.storage.getUserDetailsForUser(
+                userId
+            );
+            const links = controller.plugins.storage.getUserLinksForUser(
+                userId
+            );
+            const gauges = controller.plugins.storage.getUserGaugesForUser(
+                userId
+            );
+
+            return Promise.all([userInfo, details, links, gauges]).then(
```

Comment:
can we try to stick to a consistent syntax for working with promises in this codebase?  I'm not really a fan of mixing `.then` and `.catch` with `async / await`.  I think we should stick to one or the other, and try to avoid using both.  for example, this could be rewritten to look like this:

```js
const result = await Promise.all([userInfo, details, links, gauges]);

return {
  info: result[0],
  details: result[1],
  links: result[2],
  gauges: result[3],
};
```

I think this implementation looks cleaner and makes it easier for folks who aren't as familiar with JS / promises to understand what's going on